src/main/java/com/hivemq/mqtt/message/pool/FreePacketIdRanges.java

```diff
@@ -79,7 +79,7 @@ public int takeNextId() throws NoMessageIdAvailableException {
      *
      * @param id an ID that the caller attempts to take.
      */
-    public void takeIfAvailable(final int id) {
+    public synchronized void takeIfAvailable(final int id) {
         Preconditions.checkArgument(id >= MIN_ALLOWED_MQTT_PACKET_ID && id <= MAX_ALLOWED_MQTT_PACKET_ID,
                 "Attempting to take an ID %s that is outside the valid packet IDs range.",
                 id);
```

Comment:
I can't comment on line 92, but should we not throw there instead of return? Like in the previous implementation, when a packet ID was supposed to be taken but not available, "which should never happen"? I don't think we should silently ignore this possible error here.

Reply:
Hm, makes sense to add; I will then have to rename the method since 'IfAvailable' supposes that if it is not available, it's ok